tests/integration/framework/util/parallel.go

```diff
@@ -0,0 +1,89 @@
+/*
+Copyright 2023 The Dapr Authors
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+    http://www.apache.org/licenses/LICENSE-2.0
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package util
+
+import (
+	"sync"
+	"testing"
+
+	"github.com/stretchr/testify/assert"
+)
+
+// ParallelTest is a helper for running tests in parallel without having to
+// create new Go test functions.
+type ParallelTest struct {
+	lock sync.Mutex
+	fns  []func(*assert.CollectT)
+}
+
+// NewParallel creates a new ParallelTest.
+// Tests are executed during given test's cleanup.
+func NewParallel(t *testing.T, fns ...func(*assert.CollectT)) *ParallelTest {
+	p := &ParallelTest{
+		fns: fns,
+	}
+	t.Cleanup(func() {
+		p.lock.Lock()
+		defer p.lock.Unlock()
+		t.Helper()
+
+		jobs := make(chan func(), len(p.fns))
+		for i := 0; i < 5; i++ {
```

Comment:
probably this `5` could be an env var or some other configurable variable

Reply:
other than that lgtm

Reply:
Done :+1: